### PR TITLE
[swift-driver] Make checked build-record paths in many tests insensitive to ./ prefix

### PR DESCRIPTION
**Pull request by davidungar**
The swift-driver normalizes relative paths with the effect that an input file "./main.swift" gets written to a build record entry as "main.swift", without the "./". Remove the check of that part of the path from those tests in `Driver/Dependencies` where it is easy to do so, in service of getting the new driver to pass.

### CODE REVIEW

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Review by artemcm — APPROVED** [review]

File: test/Driver/Dependencies/check-interface-implementation-fine.swift [hunk 1]
```diff
@@ -14,9 +14,9 @@
 // CHECK-FIRST: Handled c.swift
 // CHECK-FIRST: Handled bad.swift
 
-// CHECK-RECORD-CLEAN-DAG: "./a.swift": [
-// CHECK-RECORD-CLEAN-DAG: "./bad.swift": [
-// CHECK-RECORD-CLEAN-DAG: "./c.swift": [
+// CHECK-RECORD-CLEAN-DAG: a.swift": [
```
[the remark this answers is not in the document]
Comment (davidungar) [reply test/Driver/Dependencies/check-interface-implementation-fine.swift:17]: Thank you for looking at this, @artem. IIUC, `FileCheck` is happy with partial matches, so the tests still succeed. The question in my mind is whether the extra stuff in the test is worth it to ensure that there is no bogus stuff between the opening quote and the basename. What do you think?

[the remark this answers is not in the document]
Comment (artemcm) [reply test/Driver/Dependencies/check-interface-implementation-fine.swift:17]: Oh yes, you are correct, ignore my comment :). 
I think this is fine, as-is.

[the remark this answers is not in the document]
Comment (davidungar) [reply test/Driver/Dependencies/check-interface-implementation-fine.swift:17]: Hold on, I may be wrong.

[the remark this answers is not in the document]
Comment (davidungar) [reply test/Driver/Dependencies/check-interface-implementation-fine.swift:17]: Looks like the -DAG part may not be working on one of those tests?!

[the remark this answers is not in the document]
Comment (davidungar) [reply test/Driver/Dependencies/check-interface-implementation-fine.swift:17]: OK, it turns out that the combination of `-DAG` and partial line matching is what fails. I'll go with your suggestion. Thank you!

**Comment by davidungar** [comment]
@artem , Sorry for the bogus messages! The intended person has corrected me. Please forgive me. Thanks.

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test

**Comment by davidungar** [comment]
@swift-ci please smoke test linux platform

**Pull request merged**